src/components/cta/ExternalsAuth.tsx

```diff
@@ -1,89 +1,207 @@
+      fill='white'
+    />
+    <path
+      d='M3.41846 4.83815H4.21242C4.42394 4.82434 4.63245 4.89385 4.79338 5.0318C4.85034 5.08648 4.89526 5.15243 4.92527 5.22546C4.95528 5.29849 4.96972 5.37697 4.96766 5.4559C4.97115 5.57643 4.93705 5.69507 4.87008 5.79535C4.80311 5.89563 4.70659 5.97259 4.59392 6.01555L5.0277 6.64879H4.44674L4.08268 6.09882H3.9142V6.64879H3.41846V4.83815ZM4.19306 5.70958C4.36154 5.70958 4.46223 5.62631 4.46223 5.49463C4.46223 5.34939 4.35766 5.27774 4.19112 5.27774H3.91227V5.71539L4.19306 5.70958Z'
+      fill='white'
+    />
+    <path
+      d='M5.66284 4.82275H6.14696L6.92157 6.65082H6.38128L6.25154 6.32355H5.54858L5.41884 6.65082H4.89404L5.66284 4.82275ZM6.10242 5.93237L5.90877 5.4192L5.7035 5.93237H6.10242Z'
+      fill='white'
+    />
+    <path d='M6.99316 4.83228H7.49666V6.64678H6.99316V4.83228Z' fill='white' />
+    <path
+      d='M7.69043 4.83228H8.15906L8.81167 5.80053V4.83228H9.30935V6.64678H8.8717L8.18811 5.65142V6.64678H7.69043V4.83228Z'
+      fill='white'
+    />
+    <path
+      d='M9.39453 6.37192L9.67532 6.03691C9.83849 6.17372 10.0434 6.2509 10.2563 6.25573C10.3938 6.25573 10.4654 6.20926 10.4654 6.12986C10.4654 6.0524 10.4054 6.01367 10.1575 5.95364C9.77021 5.86456 9.47393 5.75999 9.47393 5.38624C9.47393 5.04929 9.73923 4.80529 10.1749 4.80529C10.4449 4.7949 10.7098 4.88068 10.9224 5.04735L10.6784 5.40367C10.5315 5.29377 10.3544 5.23159 10.1711 5.22551C10.0491 5.22551 9.98904 5.27392 9.98904 5.3417C9.98904 5.42303 10.051 5.45983 10.3047 5.51792C10.7191 5.607 10.9708 5.74256 10.9708 6.08145C10.9708 6.45132 10.6784 6.6624 10.2369 6.6624C9.93031 6.67011 9.63122 6.56697 9.39453 6.37192Z'
+      fill='white'
+    />
+  </svg>
+);
+
+const ExternalsLink: FunctionComponent<ExternalsLinkProps> = ({
+  label,
+  dark,
+  id,
+  handleOnClick,
+  className,
+  plan = 'free',
+  icon,
+  link,
+  openInNewTab,
+}) => (
+  <Link
+    href={plan === PLAN_PRO ? `${link + '?pro=true'}` : `${link}`}
+    className={classNames(
+      'hover:sg-bg-hover-external-auth-button flex items-center justify-center px-4 py-2 font-normal hover:text-black md:h-12 md:px-6 md:text-base',
+      dark
+        ? 'hover:btn-primary bg-black text-white'
+        : 'btn-inverted-primary rounded-[3px] border border-gray-400 text-black',
+      className
+    )}
+    onClick={handleOnClick}
+    id={id}
+    {...(openInNewTab ? { target: '_blank', rel: 'noopener noreferrer' } : {})}
+  >
+    {icon}
+    {label}
+  </Link>
+);
+const logEvent = (
+  eventArguments: {
+    type: string;
+    source: string;
+    description: string;
+  },
+  eventName: string
+): void => {
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises
+  getEventLogger().log(eventName, eventArguments, eventArguments);
+};
+export const ExternalsAuth: FunctionComponent<ExternalsAuthProps> = ({
```

Comment:
So this file/component are called External**Auth**, and changing the content to no longer just be about _auth_ results in really confusing code. E.g., I had to dig around when I saw the [if statement here](https://github.com/sourcegraph/about/pull/6854/files#diff-f017a7940db2448c414a41887b377e6ef2e688a55e09f04e7e09477fa8522657R236-R239) to understand what was happening ("why is VS Code an auth provider option? Is that some new product feature?").

I don't know how much you care about code cleanliness, but if I were you I would refactor this to separate them somehow. Instead of lumping _installation_ options (VS Code and JetBrains) into a property called _authProvider_ in a component called _ExternalsAuth_ (which doesn't make much sense), I would potentially create two separate components (one for auth providers and one for installation options?) Or refactor this in some other way.

Ultimately though, if you're maintaining this, then it's your call how clean to keep this code. I don't mean to delay this release. If you would are interested in refactoring, please call in someone on Engineering (frontend, perhaps Felix K or Vova or Peter etc?) to review this change and make more specific suggestions.

Reply:
Hi @dadlerj Thanks for taking the time to provide this solid feedback. 
I appreciate your thoroughness in reviewing the code and pointing out areas where clarity could be improved.

I think I will take the approach of renaming the whole component and the different props to match the installation vs authentication.